process-document/src/main/kotlin/com/ritense/processdocument/service/DocumentDelegateService.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.processdocument.service
+
+import com.fasterxml.jackson.core.JsonPointer
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.JsonNode
+import com.ritense.authorization.AuthorizationContext
+import com.ritense.document.domain.Document
+import com.ritense.document.domain.DocumentVersion
+import com.ritense.document.domain.impl.JsonSchemaDocumentId
+import com.ritense.document.service.DocumentService
+import com.ritense.document.service.impl.JsonSchemaDocumentService
+import com.ritense.processdocument.domain.impl.CamundaProcessInstanceId
+import com.ritense.valtimo.contract.json.Mapper
+import mu.KotlinLogging
+import org.camunda.bpm.engine.delegate.DelegateExecution
+import java.time.LocalDateTime
+import java.util.*
+
+
+class DocumentDelegateService(
+    private val processDocumentService: ProcessDocumentService,
+    private val documentService: DocumentService,
+    private val jsonSchemaDocumentService: JsonSchemaDocumentService
+) {
+
+    private val mapper = Mapper.INSTANCE.get()
+
+    fun getDocumentVersion(execution: DelegateExecution): DocumentVersion? {
+        logger.debug("Get version of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).version()
```

Comment:
I think we should have tests for this. Most of it is trivial, but it's still valuable in the future.

Reply:
I agree. I will start with making these.